globus_cli/commands/endpoint/is_activated.py

```diff
@@ -0,0 +1,68 @@
+import click
+
+from globus_cli.parsing import common_options, endpoint_id_arg
+from globus_cli.safeio import safeprint
+from globus_cli.helpers import outformat_is_json, print_json_response
+from globus_cli.services.transfer import get_client
+
+
+def _print_out(text_message, response_obj):
+    if outformat_is_json():
+        print_json_response(response_obj)
+    else:
+        safeprint(text_message)
+
+
+@click.command('is-activated', short_help='Check if an Endpoint is activated',
+               help=('Check if an Endpoint is activated or requires '
+                     'activation. If it requires activation, exits with '
+                     'status 1, otherwise exits with status 0.'))
+@common_options
+@endpoint_id_arg
+@click.option('--until', type=int,
+              help=('An integer number of seconds in the future. If the '
+                    'endpoint is activated, but will expire by then, exits '
+                    'with status 1'))
+@click.option('--absolute-time', is_flag=True,
+              show_default=True, default=False,
+              help=('Treat the value of --until as a POSIX timestamp, not a '
+                    'number of seconds into the future.'))
+def endpoint_is_activated(endpoint_id, until, absolute_time):
+    """
+    Executor for `globus endpoint is-activated`
+    """
+    client = get_client()
+    res = client.endpoint_get_activation_requirements(endpoint_id)
+
+    def fail(deadline=None):
+        exp_string = ''
+        if deadline is not None:
+            exp_string = ' or will expire before {}'.format(deadline)
+
+        _print_out(('{0} is not activated{1}\n'
+                    'To activate, please go to the following page:\n\n'
+                    '  https://www.globus.org/app/endpoints/{0}/activate\n')
+                   .format(endpoint_id, exp_string), res)
+        click.get_current_context().exit(1)
+
+    def success(msg, *format_params):
+        _print_out(msg.format(endpoint_id, *format_params), res)
+        click.get_current_context().exit(0)
+
+    # autoactivation always means success, so short-circuit out of here
+    if res.supports_auto_activation:
+        success('{} does not require activation')
```

Comment:
Yes, I think "ok, I'll try and get a valid credential from Nexus" is a legitimate auto-activation operation, so that's why it always comes back as supported.